src/Npgsql/NpgsqlTypes/NpgsqlTypes.cs

```diff
@@ -461,12 +461,22 @@ public NpgsqlInet(IPAddress address)
     }
 
     public NpgsqlInet(string addr)
-        => (Address, Netmask) = addr.Split('/') switch
+    {
+        switch (addr.Split('/'))
         {
-            { Length: 2 } segments => (IPAddress.Parse(segments[0]), byte.Parse(segments[1])),
-            { Length: 1 } segments => (IPAddress.Parse(segments[0]), (byte)32),
-            _ => throw new FormatException("Invalid number of parts in CIDR specification")
-        };
+        case { Length: 2 } segments:
+            (Address, Netmask) = (IPAddress.Parse(segments[0]), byte.Parse(segments[1]));
+            break;
+        case { Length: 1 } segments:
+            var ipAddr = IPAddress.Parse(segments[0]);
+            (Address, Netmask) = (
+                ipAddr,
+                ipAddr.AddressFamily == AddressFamily.InterNetworkV6 ? (byte)128 : (byte)32);
```

Comment:
Just to be on the extra safe side, maybe have a switch expression here for the two InterNetwork options, and throw argument exception for default?

Reply:
moved AddressFamily assertion to a private function for checking type in both constructors and for raising same exception

Reply:
Pushed a commit to inline the check as a switch expression, seems a bit heavy to check the IP's type twice.